### PR TITLE
[WIP] adding hbase storage

### PR DESCRIPTION
**Pull request by HaiShin**

### CODE REVIEW

File: core/src/main/java/io/seata/core/protocol/ProtocolConstants.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
     /**
      * Max frame length
      */
-    int MAX_FRAME_LENGTH = 8 * 1024 * 1024;
+    int MAX_FRAME_LENGTH = 1024 * 1024 * 1024;
```
Comment (l81893521) [inline core/src/main/java/io/seata/core/protocol/ProtocolConstants.java:42]: Why change the frame length.

File: server/src/main/java/io/seata/server/storage/hbase/HBaseSingleConnectionFactory.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                        org.apache.hadoop.conf.Configuration configuration = HBaseConfiguration.create();
+                        configuration.set("hbase.zookeeper.quorum", CONFIGURATION.getConfig(ConfigurationKeys.STORE_HBASE_ZOOKEEPER_QUORUM));
+//                        configuration.set("hbase.zookeeper.property.clientPort", CONFIGURATION.getConfig(ConfigurationKeys.STORE_HBASE_PROPERTY_CLIENTPORT));
+                        configuration.set("hbase.client.ipc.pool.type", CONFIGURATION.getConfig(ConfigurationKeys.STORE_HBASE_POOL_TYPE,POOL_TYPE));
+                        configuration.set("hbase.client.ipc.pool.size", CONFIGURATION.getConfig(ConfigurationKeys.STORE_HBASE_POOL_SIZE,POOL_SIZE));
+                        connection = ConnectionFactory.createConnection(configuration);
+                        LOGGER.info("the connection of HBase is created successfully.");
+                    } catch (IOException e) {
+                        LOGGER.error("the connection of HBase is created failed.", e.getMessage());
+                        //改成HBase专有异常类
```
Comment (l81893521) [inline server/src/main/java/io/seata/server/storage/hbase/HBaseSingleConnectionFactory.java:51]: Please do not use Chinese comment.

Comment (HaiShin) [reply server/src/main/java/io/seata/server/storage/hbase/HBaseSingleConnectionFactory.java:51]: ok

File: core/src/main/java/io/seata/core/protocol/ProtocolConstants.java (hunk 1 above)
Comment (HaiShin) [reply core/src/main/java/io/seata/core/protocol/ProtocolConstants.java:42]: ok

File: seata-spring-autoconfigure/seata-spring-autoconfigure-server/src/main/java/io/seata/spring/boot/autoconfigure/SeataServerPropertiesAutoConfiguration.java [hunk 3]
```diff
@@ -19,28 +19,12 @@
...
-import static io.seata.spring.boot.autoconfigure.StarterConstants.METRICS_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_SESSION_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_LOCK_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_FILE_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_DB_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_REDIS_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_REDIS_SINGLE_PREFIX;
-import static io.seata.spring.boot.autoconfigure.StarterConstants.STORE_REDIS_SENTINEL_PREFIX;
+import static io.seata.spring.boot.autoconfigure.StarterConstants.*;
```
Comment (l81893521) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-server/src/main/java/io/seata/spring/boot/autoconfigure/SeataServerPropertiesAutoConfiguration.java:27]: Please do not import *

File: seata-spring-autoconfigure/seata-spring-autoconfigure-server/src/main/java/io/seata/spring/boot/autoconfigure/properties/server/store/StoreHBaseProperties.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+package io.seata.spring.boot.autoconfigure.properties.server.store;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.stereotype.Component;
+
+import static io.seata.spring.boot.autoconfigure.StarterConstants.*;
```
Comment (l81893521) [inline seata-spring-autoconfigure/seata-spring-autoconfigure-server/src/main/java/io/seata/spring/boot/autoconfigure/properties/server/store/StoreHBaseProperties.java:6]: Please do not import *

File: server/src/main/java/io/seata/server/storage/hbase/session/HBaseSessionManager.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+    }
+
+    @Override
+    public Collection<GlobalSession> allSessions() {
+        // get by taskName
+        if (SessionHolder.ASYNC_COMMITTING_SESSION_MANAGER_NAME.equalsIgnoreCase(taskName)) {
+            return findGlobalSessions(new SessionCondition(GlobalStatus.AsyncCommitting));
+        } else if (SessionHolder.RETRY_COMMITTING_SESSION_MANAGER_NAME.equalsIgnoreCase(taskName)) {
+            return findGlobalSessions(new SessionCondition(new GlobalStatus[]{
+                    GlobalStatus.CommitRetrying, GlobalStatus.Committing}));
```
Comment (l81893521) [inline server/src/main/java/io/seata/server/storage/hbase/session/HBaseSessionManager.java:173]: has 1 more \t

File: server/pom.xml [hunk 6]
```diff
@@ -52,6 +52,12 @@
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
+            <exclusions>
```
Comment (l81893521) [inline server/pom.xml:55]: Maybe this cause the unit test failed.

File: server/src/main/resources/application.yml [hunk 7]
```diff
@@ -19,12 +19,49 @@ logging:
 seata:
   config:
     # support: nacos 、 consul 、 apollo 、 zk  、 etcd3
-    type: file
+    type: nacos
```
Comment (l81893521) [inline server/src/main/resources/application.yml:22]: no need to push private configuration

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Pull request closed without merging**